File: 3 курс/01.02 поддержка и тестирование программных модулей/main.py
from datetime import datetime
from typing import List

class StudentManager:
    def min_avg(self, marks: List[str]) -> float:
        if not marks:
            raise ValueError("List of marks is empty")
        return sum(map(float, marks)) / len(marks)

    def get_count_truancy(self, marks: List["Mark"]) -> int:
        truancy_count = sum(1 for mark in marks if mark.estimation.lower() == "прогул")
        return truancy_count

    def get_count_disease(self, marks: List["Mark"]) -> int:
        disease_count = sum(1 for mark in marks if mark.estimation.lower() == "болезнь")
        return disease_count

    def get_stud_number(self, year: int, group: int, fio: str) -> str:
        fio_parts = fio.split()
        initials = "".join(part[0].upper() for part in fio_parts)
        return f"{year}.{group}.{initials}"

    def get_marks(self, now: datetime, students: List[str]) -> List["Mark"]:
        # Здесь должен быть код для генерации оценок для переданных студентов
        pass

class Mark:
    def __init__(self, date: datetime, estimation: str):
        self.date = date
        self.estimation = estimation

File: 3 курс/01.02 поддержка и тестирование программных модулей/main.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class School
{
    public double MinAVG(string[] marks)
    {
        double sum = 0;
        foreach (string mark in marks)
        {
            sum += Convert.ToDouble(mark);
        }
        return Math.Floor(sum / marks.Length);
    }

    public int[] GetCountTruancy(List<Mark> marks)
    {
        int[] counts = new int[marks.Count];
        for (int i = 0; i < marks.Count; i++)
        {
            if (marks[i].Estimation == "прогул")
            {
                counts[i]++;
            }
        }
        return counts;
    }

    public int[] GetCountDisease(List<Mark> marks)
    {
        int[] counts = new int[marks.Count];
        for (int i = 0; i < marks.Count; i++)
        {
            if (marks[i].Estimation == "болезнь")
            {
                counts[i]++;
            }
        }
        return counts;
    }

    public string GetStudNumber(int year, int group, string fio)
    {
        string initials = string.Join("", fio.Split(' ').Select(s => s[0]).ToArray());
        return $"{year}.{group}.{initials}";
    }

    public List<Mark> GetMarks(DateTime now, List<string> students)
    {
        List<Mark> marks = new List<Mark>();
        Random random = new Random();

        foreach (string student in students)
        {
            for (int i = 0; i < 10; i++)
            {
                marks.Add(new Mark
                {
                    Date = now.AddDays(i),
                    Estimation = random.Next(2) == 0 ? "прогул" : random.Next(2) == 0 ? "отсутствие" : random.Next(2) == 0 ? "болезнь" : "оценка"
                });
            }
        }

        return marks;
    }
}

public class Mark
{
    public DateTime Date { get; set; }
    public string Estimation { get; set; }
}
